packages/embark-core/src/utils.js

```diff
@@ -0,0 +1,25 @@
+import * as path from 'path';
+import {joinPath, sha512, tmpDir} from 'embark-utils';
+
+export function dappPath(...names) {
+  const DAPP_PATH = process.env.DAPP_PATH || process.cwd();
+  return path.join(DAPP_PATH, ...names);
+}
+
+export function ipcPath(basename, usePipePathOnWindows = false) {
+  if (!(basename && typeof basename === 'string')) {
+    throw new TypeError('first argument must be a non-empty string');
+  }
+  if (process.platform === 'win32' && usePipePathOnWindows) {
+    return `\\\\.\\pipe\\${basename}`;
+  }
+  return joinPath(
+    tmpDir(`embark-${sha512(dappPath()).slice(0, 8)}`),
+    basename
+  );
+}
+
+export function embarkPath(...names) {
+  const EMBARK_PATH = process.env.EMBARK_PATH;
+  path.join(EMBARK_PATH, ...names);
+}
```

Comment:
I'm pretty sure @michaelsbradleyjr has concerns to raise there. Not only because there's no fallback for `EMBARK_PATH`.

PTAL.